File: common/src/main/java/com/pluxity/permission/PermissionRepository.java
package com.pluxity.permission;

import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface PermissionRepository extends JpaRepository<Permission, Long> {

    Optional<Permission> findByResourceNameAndResourceId(String resourceName, String resourceId);

    boolean existsByResourceNameAndResourceId(String resourceName, String resourceId);

    List<Permission> findByResourceNameAndResourceIdIn(String resourceName, List<String> resourceIds);
}

File: common/src/main/java/com/pluxity/permission/PermissionRepository.kt
package com.pluxity.permission

import org.springframework.data.jpa.repository.JpaRepository

interface PermissionRepository : JpaRepository<Permission, Long> {
    fun findByResourceNameAndResourceId(
        resourceName: String,
        resourceId: String,
    ): Permission?

    fun existsByResourceNameAndResourceId(
        resourceName: String,
        resourceId: String,
    ): Boolean

    fun findByResourceNameAndResourceIdIn(
        resourceName: String,
        resourceIds: List<String>,
    ): List<Permission>
}
